regression/esbmc/github_709-2/test.desc

```diff
@@ -0,0 +1,4 @@
+CORE
+main.c
+
+^VERIFICATION FAILED$
```

Comment:
Can you also add the safe version of this regression test?

Reply:
Not sure, do you have a suggestion? This PR is about fixing a situation that has been missed (dereference inside a literal of union type). What did you have in mind?